libs/SmartSync/src/com/salesforce/androidsdk/smartsync/manager/SyncManager.java

```diff
@@ -381,30 +390,31 @@ private void syncUp(SyncState sync, SyncUpdateCallback callback) throws Exceptio
         }
 	}
 
-    private boolean isNewerThanServer(SyncUpTarget target, JSONObject record) throws JSONException, IOException {
-        final String lastModStr = record.optString(target.getModificationDateFieldName());
+    private void syncUpOneRecord(SyncUpTarget target, String soupName,
+                                 JSONObject record, SyncOptions options) throws JSONException, IOException {
 
-        if (lastModStr == null) {
-            // We didn't capture the last modified date so we can't really enforce merge mode, returning true so that we will behave like an "overwrite" merge mode
-            return true;
-        }
-        try {
-            String serverLastModStr = target.fetchLastModifiedDate(this, record);
-            if (serverLastModStr == null) {
-                // We were unable to get the last modified date from the server
-                return true;
-            }
-            long lastModifiedDate = Constants.TIMESTAMP_FORMAT.parse(lastModStr).getTime();
-            long serverLastModifiedDate = Constants.TIMESTAMP_FORMAT.parse(serverLastModStr).getTime();
-            return (serverLastModifiedDate <= lastModifiedDate);
-        } catch (Exception e) {
-            Log.e(TAG, "Couldn't figure out last modified date", e);
-            throw new SmartSyncException(e);
+        logger.d(this, "syncUpOneRecord", record);
+
+        /*
+         * Checks if we are attempting to sync up a record that has been updated
+         * on the server AFTER the client's last sync down. If the merge mode
+         * passed in tells us to leave the record alone under these
+         * circumstances, we will do nothing and return here.
+         */
+        final MergeMode mergeMode = options.getMergeMode();
+        if (mergeMode == MergeMode.LEAVE_IF_CHANGED &&
+                !target.isNewerThanServer(this, record)) {
+
+            // Nothing to do for this record
+            logger.d(this, "syncUpOneRecord: Record not synced since client does not have the latest from server:", record);
+            return;
         }
-    }
 
-    private void syncUpOneRecord(SyncUpTarget target, String soupName,
-                                 JSONObject record, SyncOptions options) throws JSONException, IOException {
+        // Advanced sync up target take it from here
+        if (target instanceof AdvancedSyncUpTarget) {
```

Comment:
ParentChildrenSyncUpTarget deals with multiple records at once, so it's no simply doing a create or update or delete. So instead of polluting the existing APIs, I introduced the new interface AdvancedSyncUpTarget. When the target implements it, sync manager simply calls syncUpRecord on the target and it should take care of sending the record(s) up to the server and updating the local store.
NB: deciding whether to skip the record or not (when merge mode is leave-if-changed) still happens in the SyncManager.